rpcs/src/methods/chain_head.rs

```diff
@@ -295,6 +291,132 @@ impl<T: RpcConfig> ChainHeadRpcMethods<T> {
             .request("transaction_v1_stop", rpc_params![operation_id])
             .await
     }
+
+    /// Fetch the block body (ie the extrinsics in the block) given its hash.
+    ///
+    /// Returns an array of the hexadecimal-encoded scale-encoded extrinsics found in the block,
+    /// or `None` if the block wasn't found.
+    pub async fn archive_unstable_body(
+        &self,
+        block_hash: T::Hash,
+    ) -> Result<Option<Vec<Bytes>>, Error> {
+        self.client
+            .request("archive_unstable_body", rpc_params![block_hash])
+            .await
+    }
+
+    /// Call the `archive_unstable_call` method and return the response.
+    pub async fn archive_unstable_call(
+        &self,
+        block_hash: T::Hash,
+        function: &str,
+        call_parameters: &[u8],
+    ) -> Result<ArchiveCallResult, Error> {
+        use serde::de::Error as _;
+
+        // We deserialize to this intermediate shape, since
+        // we can't have a boolean tag to denote variants.
+        #[derive(Deserialize)]
+        struct Response {
+            success: bool,
```

Comment:
This success flag is annoying, why do we need it at all?

result/value or error should be sufficient?

Reply:
Yeah true; I just added it as an extra check to catch any weird responses eg success:false but a value!

Reply:
isn't success part of the spec? whats the issue here

Reply:
Yep this is quite annoying in rust, IIRC we only left the success flag because it might be easier for js/ts realm interpret the data easier

Reply:
aight, I see